src/useCases/setupNewUser.ts

```diff
@@ -23,9 +25,11 @@ export default async (
   userCreateDetails: any
 ): PromiseResult<newUserSetupResult> => {
   const result = await createUser(connection, currentUser, userCreateDetails)
+  const groupsForCurrentUser = await getUserGroups(connection, [currentUser.username, userCreateDetails.username])
```

Comment:
this getUserGroups has not been moved further down. We could fail the storePasswordResetCodeor the createNewUserEmail operations

Reply:
for some reason I took it upon myself to move one line not the whole lot.

Review comment 3